cmd/watcher/list/list.go

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright 2020 VMWare.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package list
+
+import (
+	"context"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/metadata"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/urlclient/local"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+
+	"github.com/edgexfoundry/edgex-cli/config"
+	"github.com/edgexfoundry/edgex-cli/pkg/formatters"
+
+	"github.com/spf13/cobra"
+)
+
+const template = "ID\tName\tProfile\tService\tAdminState\n" +
+	"{{range .}}" +
+	"{{.Id}}\t{{.Name}}\t{{.Profile.Name}}\t{{.Service.Name}}\t{{.AdminState}}\n" +
+	"{{end}}"
+
+// NewCommand returns the list watcher command
+func NewCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "A list of all provisioning watchers",
+		Long:  `Return a list of all provisioning watchers or retrieve a provisioning watcher by id`,
+		Args:  cobra.MaximumNArgs(1),
+		RunE:  listHandler,
+	}
+	return cmd
+}
+
+func listHandler(cmd *cobra.Command, args []string) (err error) {
+	client := metadata.NewProvisionWatcherClient(
+		local.New(config.Conf.Clients["Metadata"].Url() + clients.ApiProvisionWatcherRoute),
+	)
+
+	var watchers []models.ProvisionWatcher
+	if len(args) == 0 {
+		watchers, err = client.ProvisionWatchers(context.Background())
+	} else {
+		watchers, err = getProvisioningWatcher(client, args[0])
```

Comment:
provision is a verb -- something you provide/supply/set up/give
it's past tense is what was provisioned
provisioning is what you are about to provide .

Reply:
Done